### PR TITLE
Area chart enhancements

### PR DESCRIPTION
**Pull request by hossamElfarFactorial**
## 🔑 What?
- Make the footer optional
- Add Legend, color gradient, and axis tick count to `AreaChart`

## 📸 Visual proof

SCR-20240729-khgq

### CODE REVIEW

File: lib/components/Charts/AreaChart/index.tsx [hunk 1]
```diff
@@ -30,6 +37,7 @@ export const _AreaChart = <
     xAxis,
     yAxis,
     lineType,
+    showLegend,
```
Comment (josepjaume) [inline lib/components/Charts/AreaChart/index.tsx:40]: Wondering if this should be optional or not. In the end, this design system is all about having opinions and reducing the amount of choices.

File: lib/components/Charts/AreaChart/index.tsx [hunk 2]
```diff
@@ -43,22 +51,62 @@ export const _AreaChart = <
...
+          />
+        )}
         <ChartTooltip
           cursor
-          content={<ChartTooltipContent indicator="dot" />}
+          content={<ChartTooltipContent indicator="line" />}
         />
+        <defs>
+          {areas.map((area, index) => (
+            <linearGradient id={`fill${area}`} x1="0" y1="0" x2="0" y2="1">
```
Comment (josepjaume) [inline lib/components/Charts/AreaChart/index.tsx:78]: Nice job!

File: lib/components/Charts/utils/elements.tsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ export const xAxisProps = (
   tickLine: false,
   axisLine: false,
   tickMargin: 8,
+  tickCount: config?.tickCount,
```
Comment (josepjaume) [inline lib/components/Charts/utils/elements.tsx:12]: What's the use case for this? Do we really want to make it configurable?

File: lib/components/Insights/InsightsContainer/index.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ export interface InsightsContainerProps {
     title: string
     description: string
   }
-  footer: {
+  footer?: {
```
Comment (josepjaume) [inline lib/components/Insights/InsightsContainer/index.tsx:24]: I guess it's fine if footer is optional? In any case this should change according to what @dani-moreno and Desirée end up with.

File: lib/components/Insights/InsightsContainer/index.tsx [hunk 5]
```diff
@@ -33,22 +33,24 @@ export const InsightsContainer = forwardRef<
 >(({ header, footer, children }, ref) => (
   <Card ref={ref}>
     <CardHeader>
-      <CardTitle>{header.title}</CardTitle>
       <CardDescription>{header.description}</CardDescription>
+      <CardTitle>{header.title}</CardTitle>
```
Comment (josepjaume) [inline lib/components/Insights/InsightsContainer/index.tsx:37]: Why the change of order?

**Review by josepjaume — COMMENTED** [review]
Added a few comments! Also: Could you maybe add some more storybook examples to these charts to showcase these new properties?

File: lib/components/Charts/AreaChart/index.tsx (hunk 1 above)
Comment (hossamElfarFactorial) [reply lib/components/Charts/AreaChart/index.tsx:40]: Hmmm I agree, I found some examples at figma that we can use legend and some not, so I made it optional. But since the design is not final, I will always show it for now

File: lib/components/Charts/utils/elements.tsx (hunk 3 above)
Comment (hossamElfarFactorial) [reply lib/components/Charts/utils/elements.tsx:12]: We had this use case on the employees' page, we needed to add a chart in a small place that with many y points, caused the chart to look off.

File: lib/components/Insights/InsightsContainer/index.tsx (hunk 5 above)
Comment (hossamElfarFactorial) [reply lib/components/Insights/InsightsContainer/index.tsx:37]: It's based on the designs at figma, but I will revert it since the design is not yet final.

File: lib/components/Charts/AreaChart/index.tsx (hunk 1 above)
Comment (hossamElfarFactorial) [reply lib/components/Charts/AreaChart/index.tsx]: [always show area chart legend](https://github.com/factorialco/factorial-one/pull/355/commits/dac7b65721f0e2486892a1f1dfe55624601f7013)

File: lib/components/Charts/utils/elements.tsx (hunk 3 above)
Comment (josepjaume) [reply lib/components/Charts/utils/elements.tsx:12]: Okay! Should we default it to false, perhaps?

**Review by josepjaume — APPROVED** [review]
